packages/ketcher-react/src/script/ui/state/handleHotkeysOverItem.ts

```diff
@@ -56,15 +55,42 @@ function handleOtherActions({
   dispatch(onAction(newAction))
 }
 
-function handleEraser({ editor, hoveredItem }: HandleHotkeyOverItemProps) {
-  const action = fromFragmentDeletion(
-    editor.render.ctab,
-    mapItemsToArrays(hoveredItem)
-  )
+function eraseItem({
+  editor,
+  item
+}: {
+  editor: Editor
+  item: Record<string, number[]>
+}) {
+  const action = fromFragmentDeletion(editor.render.ctab, item)
+
   editor.update(action)
   editor.hover(null)
 }
 
+function handleEraser({
+  editor,
+  hoveredItem,
+  newAction,
+  dispatch
+}: HandleHotkeyOverItemProps) {
+  const item = mapItemsToArrays(hoveredItem)
+  const itemType = Object.keys(hoveredItem)[0]
+
+  if (['atoms', 'bonds'].includes(itemType)) {
```

Comment:
I would suggest to create const for 'atoms', 'bonds', etc, as we use them in different places.